libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/GoogleFragment.java

```diff
@@ -183,16 +201,24 @@ protected void disconnectGoogleClient() {
         }
     }
 
-    protected void showAccountDialog() {
+    protected void startSignInProcess() {
         // Do nothing here.  This should be overridden by inheriting class.
     }
 
-    protected void showErrorDialog(String message) {
-        AlertDialog dialog = new AlertDialog.Builder(new ContextThemeWrapper(getActivity(), R.style.LoginTheme))
-                .setMessage(message)
-                .setPositiveButton(R.string.login_error_button, null)
-                .create();
-        dialog.show();
+    protected void finishSignUp() {
+        /* This flag might get lost when the finishSignUp is called after the fragment's
+         onSaveInstanceState was called - however it's a very rare case, since the fragment is retained across
+         config changes.  */
+        mFinished = true;
```

Comment:
I think it handles following situation
1. Device is rotated
2. Event is received and as result `finishSignUp()/finishFlow()` is called
3. The mFinished is set to true
4. commitAllowingStateLoss is called, but since the configuration change is ongoing, the state is lost
5. New Activity is created
6. Retained GoogleFragments is attached
7. It removes itself, because mFinished is true

It doesn't solve the situation when the retained GoogleFragment is destroyed by the system and recreated, but it's such a rare case, that I believe it's ok to keep it as it is.

Wdyt?

Reply:
I've added a comment to the flag.